File: api/app/Models/AccountTransaction.php
<?php
// app/Models/AccountTransaction.php

namespace App\Models;

use App\Models\Concerns\HasTenant;
use Illuminate\Database\Eloquent\Model;

class AccountTransaction extends Model
{
    use HasTenant;

    public $timestamps = false;

    protected $fillable = [
        'tenant_id', 'account_id', 'type', 'amount', 'balance_after',
        'description', 'source_type', 'source_id', 'cashbox_id',
        'payment_method', 'created_by', 'created_at',
    ];

    protected function casts(): array
    {
        return [
            'amount'        => 'decimal:2',
            'balance_after' => 'decimal:2',
            'created_at'    => 'datetime',
        ];
    }
}

File: api-net/Models/AccountTransaction.cs
namespace Liftonom.Api.Models;

public class AccountTransaction
{
    public long Id { get; set; }
    public long TenantId { get; set; }
    public long AccountId { get; set; }
    public string Type { get; set; } = "";   // debit|credit
    public decimal Amount { get; set; }
    public decimal? BalanceAfter { get; set; }
    public string? Description { get; set; }
    public string? SourceType { get; set; }
    public long? SourceId { get; set; }
    public long? CashboxId { get; set; }
    public string? PaymentMethod { get; set; }
    public long? CreatedBy { get; set; }
    public DateTime CreatedAt { get; set; }
}
